### PR TITLE
Add diagnostic for wrong where requirements separation

### PR DESCRIPTION
**Pull request by kimdv**

### CODE REVIEW

File: Sources/SwiftParser/Declarations.swift [hunk 1]
```diff
@@ -759,9 +759,21 @@ extension Parser {
         }
 
         keepGoing = self.consume(if: .comma)
+        let unexpectedBetweenBodyAndTrailingComma: RawUnexpectedNodesSyntax?
+
+        // If there's a comma, keep parsing the list.
+        // If there's a "&&", diagnose replace with a comma and keep parsing
+        if self.at(.binaryOperator) && self.currentToken.tokenText == "&&", let token = self.consume(if: .binaryOperator) {
```
Comment (ahoppen) [inline Sources/SwiftParser/Declarations.swift:766]: I think you can use the following here

```swift
if let token = self.consumeIfContextualPunctuator("&&")
```

File: Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift [hunk 2]
```diff
@@ -542,6 +542,39 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
...
+  public override func visit(_ node: GenericRequirementSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let unexpected = node.unexpectedBetweenBodyAndTrailingComma,
+      let token = unexpected.first?.as(TokenSyntax.self),
+      token.tokenKind == .binaryOperator("&&"),
+      let trailingComma = node.trailingComma,
+      trailingComma.hasError,
```
Comment (ahoppen) [inline Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift:554]:
```suggestion
      trailingComma.presence == .missing,
```

File: Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift [hunk 3]
```diff
@@ -542,6 +542,39 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
...
+      let previous = node.unexpectedBetweenBodyAndTrailingComma?.previousToken(viewMode: .sourceAccurate)
+    {
+
+      addDiagnostic(
+        unexpected,
+        .expectedCommaInWhereClause,
+        fixIts: [
+          FixIt(
+            message: ReplaceTokensFixIt(replaceTokens: [TokenSyntax(token)!], replacement: .commaToken()),
```
Comment (ahoppen) [inline Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift:563]: Why do you need to do `TokenSyntax(token)` here? Isn’t `token` already of type `TokenSyntax`?

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci pleas test windows

**Comment by kimdv** [comment]
@swift-ci please test windows

**Pull request merged**